pinot-core/src/main/java/org/apache/pinot/core/common/DataFetcher.java

```diff
@@ -425,34 +426,14 @@ void readIntValues(int[] docIds, int length, int[] valueBuffer) {
         _reader.readDictIds(docIds, length, dictIdBuffer, readerContext);
         _dictionary.readIntValues(dictIdBuffer, length, valueBuffer);
       } else {
-        switch (_reader.getValueType()) {
-          case INT:
-            for (int i = 0; i < length; i++) {
-              valueBuffer[i] = _reader.getInt(docIds[i], readerContext);
-            }
-            break;
-          case LONG:
-            for (int i = 0; i < length; i++) {
-              valueBuffer[i] = (int) _reader.getLong(docIds[i], readerContext);
-            }
-            break;
-          case FLOAT:
-            for (int i = 0; i < length; i++) {
-              valueBuffer[i] = (int) _reader.getFloat(docIds[i], readerContext);
-            }
-            break;
-          case DOUBLE:
-            for (int i = 0; i < length; i++) {
-              valueBuffer[i] = (int) _reader.getDouble(docIds[i], readerContext);
-            }
-            break;
-          case STRING:
-            for (int i = 0; i < length; i++) {
-              valueBuffer[i] = Integer.parseInt(_reader.getString(docIds[i], readerContext));
-            }
-            break;
-          default:
-            throw new IllegalStateException();
+        if (_reader.getValueType().isFixedWidth()) {
```

Comment:
(Optional) Suggest including `STRING` in the reader interface for completeness. Check `isFixedWidth()` in the `FixedByteChunkSVForwardIndexReader` override method.

Reply:
I didn't add string because the optimisation isn't relevant because of the way the data's laid out. I can add it with a default implementation if you think that's cleaner.

Reply:
Yes, that is what I meant. Add `STRING` as the default implementation so that the default implementation supports all data types. In the override method we can check `isFixedWidth()` so that `STRING` can fall back to the default implementation

Reply:
I will apply this change before this gets merged